### PR TITLE
Add terraform module

### PR DESCRIPTION
**Pull request by tibor-szalai**
Add terraform module

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
+data "aws_partition" "current" {}
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "doit-assume-role-policy" {
```
Comment (migueldelucasdoit) [inline terraform/main.tf]: @tibor-szalai please observe naming conventions for resource names. "doit-assume-role-policy" -> "doit_assume_role_policy"
https://www.terraform-best-practices.com/naming

File: terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = [var.account_id]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "doit-iam-policy-document" {
```
Comment (migueldelucasdoit) [inline terraform/main.tf]: @tibor-szalai please observe naming conventions for resource names. "doit-assume-role-policy" -> "doit_assume_role_policy"
https://www.terraform-best-practices.com/naming

File: terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/oidc.aws-ops.doit-intl.com*",
+      "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:policy/DoiT-*",
+      "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/DoiT-Support*"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "doit-iam-policy" {
```
Comment (migueldelucasdoit) [inline terraform/main.tf]: @tibor-szalai please observe naming conventions for resource names. "doit-assume-role-policy" -> "doit_assume_role_policy"
https://www.terraform-best-practices.com/naming

File: terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  name   = "DoiT_Customer_Policy_DO-NOT-DELETE"
+  policy = data.aws_iam_policy_document.doit-iam-policy-document.json
+}
+
+resource "aws_iam_role" "doit-management-role" {
```
Comment (migueldelucasdoit) [inline terraform/main.tf]: @tibor-szalai please observe naming conventions for resource names. "doit-assume-role-policy" -> "doit_assume_role_policy"
https://www.terraform-best-practices.com/naming

File: terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  name               = "DoiT_Management_Role_DO-NOT-DELETE"
+  assume_role_policy = data.aws_iam_policy_document.doit-assume-role-policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "doit-management-role-attachment" {
```
Comment (migueldelucasdoit) [inline terraform/main.tf]: @tibor-szalai please observe naming conventions for resource names. "doit-assume-role-policy" -> "doit_assume_role_policy"
https://www.terraform-best-practices.com/naming

File: terraform/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+## Usage
```
Comment (migueldelucasdoit) [inline terraform/README.md:1]: @tibor-szalai How did you create this documentation? We have incorporated pre-commit git hooks that will help with linting, formatting, and generating documentation for Terraform code.
Please check https://github.com/antonbabenko/pre-commit-terraform

File: terraform/variables.tf [hunk 7]
```diff
@@ -0,0 +1,11 @@
+variable "account_id" {
```
Comment (migueldelucasdoit) [inline terraform/variables.tf:1]: @tibor-szalai I would not include any default value if this is a required variable as seems to be the case

**Review by migueldelucasdoit — CHANGES_REQUESTED** [review]

File: terraform/README.md (hunk 6 above)
Comment (tibor-szalai) [reply terraform/README.md:1]: Documentation has been recreated with pre-commit git hooks

File: terraform/main.tf (hunk 1 above)
Comment (tibor-szalai) [reply terraform/main.tf]: Fixed

File: terraform/main.tf (hunk 2 above)
Comment (tibor-szalai) [reply terraform/main.tf]: Fixed

File: terraform/main.tf (hunk 3 above)
Comment (tibor-szalai) [reply terraform/main.tf]: Fixed

File: terraform/main.tf (hunk 4 above)
Comment (tibor-szalai) [reply terraform/main.tf]: Fixed

File: terraform/main.tf (hunk 5 above)
Comment (tibor-szalai) [reply terraform/main.tf]: Fixed

File: terraform/variables.tf (hunk 7 above)
Comment (tibor-szalai) [reply terraform/variables.tf:1]: You're right

File: terraform/outputs.tf [hunk 8]
```diff
@@ -0,0 +1,9 @@
+output "iam_role_name" {
+  description = "The name of the IAM Role"
+  value       = aws_iam_role.doit-management-role.name
```
Comment (migueldelucasdoit) [inline terraform/outputs.tf]: @tibor-szalai Don't forget to change the outputs. ;)
```suggestion
  value       = aws_iam_role.doit_management_role.name
```

File: terraform/outputs.tf [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+}
+
+output "iam_role_arn" {
+  description = "The arn of the IAM Role"
+  value       = aws_iam_role.doit-management-role.arn
```
Comment (migueldelucasdoit) [inline terraform/outputs.tf]: @tibor-szalai Don't forget to change the outputs. ;)
```suggestion
  value       = aws_iam_role.doit_management_role.arn
```

**Review by migueldelucasdoit — CHANGES_REQUESTED** [review]

File: terraform/outputs.tf (hunk 8 above)
Comment (tibor-szalai) [reply terraform/outputs.tf]: I did forget. Thanks for reminding me. :)

File: terraform/outputs.tf (hunk 9 above)
Comment (tibor-szalai) [reply terraform/outputs.tf]: I did forget. Thanks for reminding me. :)

**Review by migueldelucasdoit — APPROVED** [review]

**Pull request merged**